oaknational/oak-ai-lesson-assistant
# fix: upgrade typescript + prettier + eslint with a single shared linting config

## Description

- Upgrades Typescript to latest
- Upgrades Eslint to latest
- Moves all config to a shared eslint-config package
- Uses the new Eslint config format

We primarily use Typescript, some ESM, and an occasional CommonJS file. So this makes TS the preferred format, ESM the secondary and CommonJS a tertiary format.

To write JS code, name your file with the .mjs extension.
To write code with CommonJS you will need to use the .cjs extension.

Linting rules will now apply to each type of file, and we can specify rules that apply generally, or specifically for certain types of file.

## How to review

- Pull the branch down locally
- Restart your editor
- Make changes to a file / break something
- Make sure that linting is performant and you get feedback as expected
- Fix the issue
- Make sure you get feedback quickly that the error is resolved
